app/controllers/posts_controller.rb

```diff
@@ -14,4 +15,26 @@ def show
     # used in the view to display the details of the post.
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  ##
+  # creates a new post associated with a user and redirects to the user's page
+  # if successful, or renders the new post form with an error message if unsuccessful.
+  def create
+    @author = User.find(params[:author_id])
+    @post = @author.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_path(id: @post.author_id)
+    else
+      render :new, alert: 'An error has occurred while creating the post'
+    end
+  end
```

Comment:
- [ ] There is a problem with creating a new post. According to the project's requirements, the post should be created for the `current_user` which we have defined in the `application.rb` file.
Implementing this would give you something similar to this 👇 
```ruby
    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```
#### Error encountered while creating post
![image](https://github.com/Marcraphael12/rails-blog-app/assets/46839548/bd01edd3-e462-4f7e-94a0-3c21818973ca)
### NB: The next reviewer will check to see that a comment and like can be added to a post when you fix this bug 🐛

Reply:
thanks for the hint... i'll apply that